src/main/java/htsjdk/samtools/util/EdgeReadIterator.java

```diff
@@ -0,0 +1,227 @@
+                        refOffsetEndInterval = refOffsetEndInterval - (refPos + alignmentBlock.getLength() - intervalEnd) + 1;
+                    }
+                }
+
+            }
+            int length = refOffsetEndInterval - refOffsetInterval;
+            // add the alignment block to the accumulator when it starts and when it ends 
+            EdgingRecordAndOffset recordAndOffset = createRecordAndOffset(rec, readOffset + startShift, length, refPos + startShift, EdgingRecordAndOffset.Type.BEGIN);
+            // accumulate start of the alignment block
+            accumulator.get(refOffsetInterval).add(recordAndOffset);
+            EdgingRecordAndOffset recordAndOffsetEnd = createRecordAndOffset(rec, readOffset + startShift, length, refPos + startShift, EdgingRecordAndOffset.Type.END);
+            recordAndOffsetEnd.setStart(recordAndOffset);
+            // accumulate end of the alignment block
+            accumulator.get(refOffsetEndInterval).add(recordAndOffsetEnd);
+        }
+    }
+
+    @Override
+    void accumulateIndels(SAMRecord rec) {
+        throw new UnsupportedOperationException("Indels accumulation is not supported for " + getClass().getSimpleName() + ".");
+    }
+
+    /**
+     * Creates a new <code>EdgingRecordAndOffset</code> for given input values
+     *
+     * @param rec        aligned SamRecord
+     * @param readOffset offset from start of read
+     * @param length     length of alignment block
+     * @param refPos     position in the reference sequence
+     * @param type       BEGIN or END type of RecordAndOffset
+     * @return created <code>EdgingRecordAndOffset</code>
+     */
+    @Override
+    EdgingRecordAndOffset createRecordAndOffset(SAMRecord rec, int readOffset, int length, int refPos, EdgingRecordAndOffset.Type type) {
+        return new EdgingRecordAndOffset(rec, readOffset, length, refPos, type);
+    }
+
+    /**
+     * @param referenceSequence processed reference sequence
+     * @param lastPosition      last processed reference locus position
+     * @return <code>AbstractLocusInfo<T></code> for the lastPosition
+     */
+    @Override
+    AbstractLocusInfo<EdgingRecordAndOffset> createLocusInfo(SAMSequenceRecord referenceSequence, int lastPosition) {
+        return new AbstractLocusInfo<>(referenceSequence, lastPosition);
+    }
+
+    /**
+     * This method isn't supported in current implementation.
+     *
+     * @param maxReadsToAccumulatePerLocus maximum number of <code>RecordAndOffset</code> objects to store for
+     *                                     one loci in reference sequence.
+     */
+    @Override
+    public void setMaxReadsToAccumulatePerLocus(int maxReadsToAccumulatePerLocus) {
+        if (getMaxReadsToAccumulatePerLocus() != maxReadsToAccumulatePerLocus) {
+            throw new UnsupportedOperationException("Locus cap is not supported for " + getClass().getSimpleName() + ".");
```

Comment:
I'm confused. We can set a Cap (as per the picard branch that does this) so what's this about?